Core/HLE/sceNetInet.cpp

```diff
@@ -457,40 +457,50 @@ static int sceNetInetSetsockopt(int socket, int level, int optname, u32 optvalPt
-		return hleLogSuccessI(Log::sceNet, 0);
-	}
-	// TODO: Ignoring SO_NOSIGPIPE flag to prevent crashing PPSSPP (not sure if PSP has NOSIGPIPE or not tho, defined as 0x1022 on Darwin)
-	else if (level == PSP_NET_INET_SOL_SOCKET && optname == 0x1022) { // PSP_NET_INET_SO_NOSIGPIPE ?
-		//memcpy(&sock->nosigpipe, (int*)optval, std::min(sizeof(sock->nosigpipe), optlen));
-		return hleLogSuccessI(Log::sceNet, 0);
-	}
-	// It seems UNO game will try to set socket buffer size with a very large size and ended getting error (-1), so we should also limit the buffer size to replicate PSP behavior
-	else if (level == PSP_NET_INET_SOL_SOCKET && (optname == PSP_NET_INET_SO_RCVBUF || optname == PSP_NET_INET_SO_SNDBUF)) { // PSP_NET_INET_SO_NOSIGPIPE ?
-		// TODO: For SOCK_STREAM max buffer size is 8 Mb on BSD, while max SOCK_DGRAM is 65535 minus the IP & UDP Header size
-		if (optval > 8 * 1024 * 1024) {
-			UpdateErrnoFromHost(ENOBUFS, __FUNCTION__); // FIXME: return ENOBUFS for SOCK_STREAM, and EINVAL for SOCK_DGRAM
-			return hleLogError(Log::sceNet, -1, "buffer size too large?");
+
+	if (level == PSP_NET_INET_SOL_SOCKET) {
+		if (optname == PSP_NET_INET_SO_ONESBCAST) {
+			// We can't set this, at least not on Windows.
+			// This occurs in Outrun 2006, during account registration and probably other processes.
+			return hleLogWarning(Log::sceNet, 0, "PSP_NET_INET_SO_ONESBCAST unsupported, ignoring");
+		}
+
+		// TODO: Ignoring SO_NBIO/SO_NONBLOCK flag if we always use non-blocking mode (ie. simulated blocking mode)
+		else if (optname == PSP_NET_INET_SO_NBIO) {
+			inetSock->nonblocking = optval;
+			return hleLogSuccessI(Log::sceNet, 0);
+		}
+		// FIXME: Should we ignore SO_BROADCAST flag since we are using fake broadcast (ie. only broadcast to friends),
+		//        But Infrastructure/Online play might need to use broadcast for SSDP and to support LAN MP with real PSP
+		/*else if (level == PSP_NET_INET_SOL_SOCKET && optname == PSP_NET_INET_SO_BROADCAST) {
+			//memcpy(&sock->so_broadcast, (int*)optval, std::min(sizeof(sock->so_broadcast), optlen));
+			return hleLogSuccessI(Log::sceNet, 0);
+		}*/
+		// TODO: Ignoring SO_REUSEADDR flag to prevent disrupting multiple-instance feature
+		else if (optname == PSP_NET_INET_SO_REUSEADDR) {
+			//memcpy(&sock->reuseaddr, (int*)optval, std::min(sizeof(sock->reuseaddr), optlen));
+			return hleLogSuccessI(Log::sceNet, 0);
+		}
+		// TODO: Ignoring SO_REUSEPORT flag to prevent disrupting multiple-instance feature (not sure if PSP has SO_REUSEPORT or not tho, defined as 15 on Android)
+		else if (optname == PSP_NET_INET_SO_REUSEPORT) { // 15
+			//memcpy(&sock->reuseport, (int*)optval, std::min(sizeof(sock->reuseport), optlen));
+			return hleLogSuccessI(Log::sceNet, 0);
+		}
+		// TODO: Ignoring SO_NOSIGPIPE flag to prevent crashing PPSSPP (not sure if PSP has NOSIGPIPE or not tho, defined as 0x1022 on Darwin)
+		else if (optname == 0x1022) { // PSP_NET_INET_SO_NOSIGPIPE ?
```

Comment:
We should probably replace this raw number with an identifier for easier to replace later if this is wrong, but i'm not sure whether there are games that use this option name or not, just a precaution not to change NOSIGPIPE on the host side.

Reply:
I'll turn it into a hleLogWarning for now. I don't think turning on signals make sense for any psp game anyway since the PSP doesn't really have unix signals...